src/cpp/desktop/DesktopOptions.cpp

```diff
@@ -151,64 +259,29 @@ void Options::saveMainWindowBounds(QMainWindow* win)
 
 QString Options::portNumber() const
 {
-#ifndef RSTUDIO_PACKAGE_BUILD
-   // allow for hard-coded port, for debugging
-   std::string port = core::system::getenv("RSTUDIO_RSESSION_PORT");
-   if (!port.empty())
-   {
-      portNumber_ = QString::fromStdString(port);
-      return portNumber_;
-   }
-#endif
-
    // if we already have a port number, use it
    if (!portNumber_.isEmpty())
    {
       LOG_DEBUG_MESSAGE("Using port: " + portNumber_.toStdString());
       return portNumber_;
    }
-   
-   // otherwise, try to find an open port
-   for (int i = 0; i < 100; i++)
-   {
-      // generate a port number
-      // RFC 6335 suggests the range 49152-65535 for dynamic ports
-      int port = core::random::uniformRandomInteger<int>(49152, 65535);
-      LOG_DEBUG_MESSAGE("Trying port " + std::to_string(port));
 
-      // try to bind to it -- if this fails, try a new port number
-      QTcpSocket socket;
-      if (!socket.bind(port))
-      {
-         LOG_DEBUG_MESSAGE("Couldn't bind to port " + std::to_string(port) + "; trying new port");
-         continue;
-      }
+   // compute the port number to use
+   DLOGF("Finding open port for communication with rsession");
+   int port = findOpenPort();
 
-      // we found a good port; close our socket and save it
-      //
-      // TODO: it'd be nice to keep the port locked until we are
-      // officially ready to launch the rsession, or let rsession
-      // communicate to us which port it wants to use via some
-      // separate channel
-      LOG_DEBUG_MESSAGE("Successfully bound to port " + std::to_string(port) + "; saving");
-      socket.close();
-      portNumber_ = QString::number(port);
-
-      // recalculate the local peer and set RS_LOCAL_PEER so that
-      // rsession and it's children can use it
+   DLOGF("Using port: {}", port);
+   portNumber_ = QString::number(port);
+
+   // recalculate the local peer and set RS_LOCAL_PEER so that
+   // rsession and its children can use it
 #ifdef _WIN32
-      QString localPeer =  QStringLiteral("\\\\.\\pipe\\%1-rsession").arg(portNumber_);
-      localPeer_ = localPeer.toUtf8().constData();
-      core::system::setenv("RS_LOCAL_PEER", localPeer_);
+   QString localPeer =  QStringLiteral("\\\\.\\pipe\\%1-rsession").arg(portNumber_);
+   localPeer_ = localPeer.toUtf8().constData();
+   core::system::setenv("RS_LOCAL_PEER", localPeer_);
 #endif
-    
-      // return discovered port
-      return portNumber_;
-   }
 
-   // if we get here, we failed to find an open port
```

Comment:
Are we sure we want to pick a random port in this fallthrough case versus returning the formerly hardcoded 8789? I have no idea if this is a good idea or not, or if anything relies on this old behaviour, but would hate to find out after the fact that somebody, somewhere, was depending on this.

Reply:
That was a change introduced with https://github.com/rstudio/rstudio/commit/d07d88465423fe224d86e3337f531ebfb90b505a (that is, defaulting to 8789 wasn't legacy behavior) so I think this change actually brings us back closer to the older behavior.

However, it might be worth considering -- we previously considered ports in the range 4000 through 8000. Should we consider that range as a fallback? (assuming some existing environments are expecting RStudio + rsession to communicate on a port in that range?)